src/metric/base.ts

```diff
@@ -17,6 +17,8 @@ limitations under the License.
 import { Dimension, Metric } from "./types";
 
 export abstract class BaseMetric<T> implements Metric {
+    private static metricLineMaxLength = 50_000;
```

Comment:
nit: static class members can't easily be minified. it is better to have a separate constant. Also uppercase names is the convention for constants so people know not to overwrite them.

```js
const METRIC_LINE_MAX_LENGTH = 50_000
```